src/porepy/numerics/mixed_dim/coupler.py

```diff
@@ -101,13 +112,24 @@ def split(self, gb, key, values):
 
         """
         dofs = np.empty(gb.size(), dtype=int)
-        for _, d in gb:
-            dofs[d['node_number']] = d['dof']
+        for _, data in gb:
+            k_data = self.__get_data(data, data_key)
+            dofs[data['node_number']] = k_data['dof']
         dofs = np.r_[0, np.cumsum(dofs)]
 
-        gb.add_node_prop(key)
+        if data_key is None:
+            gb.add_node_prop(key)
+
         for g, d in gb:
-            i = d['node_number']
-            d[key] = values[slice(dofs[i], dofs[i+1])]
+            k_data = self.__get_data(data, data_key)
+            i = data['node_number']
+            k_data[key] = values[slice(dofs[i], dofs[i+1])]
 
 #------------------------------------------------------------------------------#
+    def __get_data(self, data, data_key):
```

Comment:
I've not though about the case where you need more than one subdict within a solver. I guess you could have one key-word per subdict (`Biot().discretize(g, data, flow_key_word='flow', mech_key_word='mech', compr_key_work='compr')`), but it starts to become a bit cumbersome. 

The transition to a single grid is quite simple:
`
    g = gb.grids_of_dimension(3)[0]
    
    data = gb.graph.node[g]
    A_e, b_e = solver_elasticity.matrix_rhs(g, data['elasticity'])
    sol = sps.linalg.spsolve(A_e.copy(), b_e)
`